src/data/thunks.js

```diff
@@ -340,12 +341,19 @@ export function expireExam() {
  * @param workerUrl - location of the worker from the provider
  */
 export function pingAttempt(timeoutInSeconds, workerUrl) {
-  return async (dispatch) => {
+  return async (dispatch, getState) => {
     await pingApplication(timeoutInSeconds, workerUrl)
-      .catch((error) => handleAPIError(
-        { message: error ? error.message : 'Worker failed to respond.' },
-        dispatch,
-      ));
+      .catch(async (error) => {
+        const { exam, activeAttempt } = getState().examState;
+        const message = error ? error.message : 'Worker failed to respond.';
+        await updateAttemptAfter(
+          exam.course_id, exam.content_id, endExamWithFailure(activeAttempt.attempt_id, message),
+        )(dispatch);
+        handleAPIError(
```

Comment:
do we need handleApiError here? I think after you end exam with failure you already show error instructions page so you don't need to add it to state also

Reply:
I agree with you. Changed.